filestore/s3.go

```diff
@@ -65,7 +64,7 @@ func NewS3FileStore(
 		// Ignore for now.
 		return nil, err
 	}
-	return &S3FileStore{s3client: s3client, minioClient: minioClient, bucket: bucket, disableSSLverify: disableSSLverify}, nil
+	return &S3FileStore{s3client: s3client, minioClient: minioClient, bucket: bucket, httpClient: httpClient}, nil
```

Comment:
Should check that the httpClient isn't `nil` and make a test for that case

Reply:
See examples above for the other 2 clients

Reply:
Check for nil. Haven't added the test yet.

Reply:
Test added to s3_test.go; made some changes in constructFail to accomodate.